src/overlays/actors/ovl_En_Test3/z_en_test3.c

```diff
@@ -27,113 +203,1114 @@ const ActorInit En_Test3_InitVars = {
+        }
+    } else {
+        func_80128B74(play, &this->player, limbIndex);
+    }
+}
+
+static TexturePtr sEyeTextures[] = {
+    object_test3_Tex_000DC0, object_test3_Tex_003680, object_test3_Tex_003E80, object_test3_Tex_004680,
+    object_test3_Tex_004E80, object_test3_Tex_005680, object_test3_Tex_005E80, object_test3_Tex_006680,
+};
+
+static TexturePtr sMouthTextures[] = {
+    object_test3_Tex_0009C0,
+    object_test3_Tex_006E80,
+    object_test3_Tex_007280,
+    object_test3_Tex_007680,
+};
+
+typedef struct {
+    /* 0x0 */ u8 eyeIndex;
+    /* 0x1 */ u8 mouthIndex;
+} KafeiFace; // size = 0x2
+
+static KafeiFace sFaceExpressions[] = {
+    { 0, 0 }, { 1, 0 }, { 2, 0 }, { 0, 0 }, { 1, 0 }, { 2, 0 }, { 4, 0 }, { 5, 1 }, { 7, 2 }, { 0, 2 },
+    { 3, 0 }, { 4, 0 }, { 2, 2 }, { 1, 1 }, { 0, 2 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 }, { 0, 0 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A40824.s")
+void EnTest3_Draw(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    EnTest3* this = THIS;
+    s32 eyeTexIndex = (this->player.skelAnime.jointTable[OBJECT_TEST3_LIMB_MAX].x & 0xF) - 1;
+    s32 mouthTexIndex = ((this->player.skelAnime.jointTable[OBJECT_TEST3_LIMB_MAX].x >> 4) & 0xF) - 1;
+    Gfx* gfx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A4084C.s")
+    OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A40908.s")
+    func_8012C268(play);
+    if (this->player.invincibilityTimer > 0) {
+        s32 temp2; // Must exist for stack order. Could hold the result of CLAMP instead.
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A409D4.s")
+        this->player.unk_B5F += CLAMP(50 - this->player.invincibilityTimer, 8, 40);
+        temp2 = Math_CosS(this->player.unk_B5F * 0x100) * 2000.0f;
+        POLY_OPA_DISP = Gfx_SetFog(POLY_OPA_DISP, 255, 0, 0, 0, 0, 4000 - temp2);
+    }
+    func_800B8050(&this->player.actor, play, 0);
+    D_80A418C8 = false;
+    if (this->player.stateFlags1 & PLAYER_STATE1_100000) {
+        Vec3f sp4C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A40A6C.s")
+        SkinMatrix_Vec3fMtxFMultXYZ(&play->viewProjectionMtxF, &this->player.actor.focus.pos, &sp4C);
+        if (sp4C.z < -4.0f) {
+            D_80A418C8 = true;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/EnTest3_Update.s")
+    gfx = POLY_OPA_DISP;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Test3/func_80A40CF0.s")
+    if (eyeTexIndex < 0) {
+        eyeTexIndex = sFaceExpressions[this->player.actor.shape.face].eyeIndex;
+    }
+    gSPSegment(&gfx[0], 8, Lib_SegmentedToVirtual(sEyeTextures[eyeTexIndex]));
+    if (mouthTexIndex < 0) {
+        mouthTexIndex = sFaceExpressions[this->player.actor.shape.face].mouthIndex;
+    }
+    gSPSegment(&gfx[1], 9, Lib_SegmentedToVirtual(sMouthTextures[mouthTexIndex]));
```

Comment:
```suggestion
    gSPSegment(&gfx[0], 0x08, Lib_SegmentedToVirtual(sEyeTextures[eyeTexIndex]));
    if (mouthTexIndex < 0) {
        mouthTexIndex = sFaceExpressions[this->player.actor.shape.face].mouthIndex;
    }
    gSPSegment(&gfx[1], 0x09, Lib_SegmentedToVirtual(sMouthTextures[mouthTexIndex]));
```
Just following convention